index.d.ts

```diff
@@ -46,4 +46,4 @@ console.log(cleanStack(error.stack));
 //     at Object.<anonymous> (/Users/sindresorhus/dev/clean-stack/unicorn.js:2:15)
 ```
 */
-export default function cleanStack(stack: string, options?: Options): string;
+export default function cleanStack<T extends string | undefined>(stack: T, options?: Options): T extends string ? string : undefined;
```

Comment:
Shouldn't it be:
```suggestion
export default function cleanStack<T extends string | undefined>(stack: T, options?: Options): T;
```

Reply:
@sindresorhus No. If the provided `stack` parameter has a type narrower than `string`, such as `'foobar'` or maybe `string & { __some_opaque_marker__: 'foobar' }`, the output of `cleanStack` should get widened to `string` instead of remaining narrow.